client/landing/stepper/declarative-flow/ai-site-builder.ts

```diff
@@ -0,0 +1,153 @@
+			switch ( currentStep ) {
+				// The create-site step will start creating a site and will add the promise of that operation to pendingAction field in the store.
+				case 'create-site': {
+					// Go to the processing step and pass `true` to remove it from history. So clicking back will not go back to the create-site step.
+					return navigate( 'processing', undefined, true );
+				}
+				// The processing step will wait the aforementioned promise to be resolved and then will submit to you whatever the promise resolves to.
+				// Which will be the created site { "siteId": "242341575", "siteSlug": "something.wordpress.com", "goToCheckout": false, "siteCreated": true }
+				case 'processing': {
+					const { siteSlug, siteId } = providedDependencies;
+
+					try {
+						await wpcom.req.post( {
+							apiNamespace: 'wpcom/v2',
+							path: `/sites/${ siteId }/send-email-continue-site-build`,
+							body: {
+								continue_url: `https://${ siteSlug }/wp-admin/site-editor.php?canvas=edit`,
+							},
+						} );
+					} catch ( error ) {
+						// eslint-disable-next-line no-console
+						console.error( 'Failed to send continue build email:', error );
+					}
+
+					// get the prompt from the get url
+					const prompt = new URLSearchParams( window.location.search ).get( 'prompt' );
+					let promptParam = '';
+
+					if ( prompt ) {
+						promptParam = `&prompt=${ prompt }`;
+					}
+
+					return navigate(
+						`launch-big-sky?siteId=${ siteId }&siteSlug=${ siteSlug }&referrer=ai-site-builder${ promptParam }`,
+						undefined,
+						true
+					);
+				}
+				case 'domains': {
+					// eslint-disable-next-line no-console
+					console.log( 'DOMAAAINZ STEP', providedDependencies, siteSlugFromSiteData );
+					// TODO: Somehow store the chosen domain.
+					if ( providedDependencies.domainItem && siteSlugFromSiteData ) {
+						const addToCart = await addProductsToCart( siteSlugFromSiteData, AI_SITE_BUILDER_FLOW, [
+							providedDependencies.domainItem,
+						] );
+						// eslint-disable-next-line no-console
+						console.log( 'ADD TO CART', addToCart );
+					}
+					return navigate( 'plans' );
+				}
+
+				case 'plans': {
+					const { cartItems } = providedDependencies;
+					// eslint-disable-next-line no-console
+					console.log(
+						'PLAAANZ STEP',
+						JSON.stringify(
+							{
+								dependencies: providedDependencies,
+								slug: siteSlugFromSiteData,
+								domain: domainCartItem,
+							},
+							null,
+							2
+						)
+					);
+
+					if ( cartItems && cartItems[ 0 ] && siteSlugFromSiteData ) {
+						const addToCart = await addPlanToCart(
+							siteSlugFromSiteData,
+							AI_SITE_BUILDER_FLOW,
+							true,
+							'assembler',
+							cartItems[ 0 ]
+						);
+						// eslint-disable-next-line no-console
+						console.log( 'ADD TO CART', addToCart );
+					}
+
+					// eslint-disable-next-line no-console
+					window.location.replace(
```

Comment:
You can use `window.location.assign`, otherwise clicking back from checkout will take the user to domains step.